### PR TITLE
MPI working on Windows

### PR DESCRIPTION
**Pull request by krystophny**
Following instructions in #194 and https://github.com/dev-cafe/autocmake/issues/85 now the MPI tests run on Windows with these modifications. If this doesn't break other stuff, please merge.

### CODE REVIEW

File: pyccel/codegen/utilities.py [hunk 1]
```diff
@@ -110,7 +115,14 @@ def compile_fortran(filename, compiler, flags,
     m_code = ' '.join('{}.o'.format(m) for m in modules)
     libs_flags = ' '.join('-l{}'.format(i) for i in libs)
 
-    cmd = '{0} {1} {2} "{3}" {4} "{5}" {6} {7} {8}'.format( \
+    filename = '"{}"'.format(filename)  # in case of spaces in path
+    binary = '"{}"'.format(binary)
+
+    if sys.platform == "win32" and compiler == "mpif90":
+        compiler = "gfortran"
```
Comment (EmilyBourne) [inline pyccel/codegen/utilities.py:122]: Why do we change the selected compiler? If this is a default then shouldn't the switch be done when the compiler is selected? If not then shouldn't we use the compiler that the user asked for?

File: pyccel/codegen/utilities.py [hunk 2]
```diff
@@ -59,6 +60,10 @@ def construct_flags(compiler,
     if compiler == "mpif90":
         if debug:
             flags += " -fbounds-check"
+        if sys.platform == "win32":
+            mpiinc = os.environ["MSMPI_INC"].rstrip('\\')
+            mpilib = os.environ["MSMPI_LIB64"].rstrip('\\')
```
Comment (EmilyBourne) [inline pyccel/codegen/utilities.py:65]: Are these environment variables specific to a certain compiler?

File: pyccel/codegen/utilities.py (hunk 1 above)
Comment (krystophny) [reply pyccel/codegen/utilities.py:122]: The problem here is that `mpif90` doesn't exist on Windows. One rather writes `gfortran` with some command line argument telling it to link MPI libraries. This solution here is not very clean, as it doesn't distinguish whether the user explicitly wants `mpif90`, and whether MPI default compiler was selected. This should indeed be changed.

File: pyccel/codegen/utilities.py (hunk 2 above)
Comment (krystophny) [reply pyccel/codegen/utilities.py:65]: This is for MPI on Windows for any compiler, since Microsoft MPI is needed there.

Comment (EmilyBourne) [reply pyccel/codegen/utilities.py:65]: From some googling it seems that there is also an Intel MPI for windows. However if the user wishes to use this I think we can suppose that they will specify their own flags. In this case would we end up with multiple MPI routines specified?

If this isn't compiler specific then why is it in the `if compiler == "mpif90":`? Is this once again related to the default compiler? If the user specified their own compiler then won't the MPI flags be missed?

Comment (yguclu) [reply pyccel/codegen/utilities.py:65]: Depending on which MPI library and compiler the user wants to use, I think that `mpif90` should point to the correct compiler. Now the question is: how do we know if the user wants to use the Intel MPI and compiler?

Comment (yguclu) [reply pyccel/codegen/utilities.py:65]: Sorry, I just read in the comment below that the `mpif90` wrapper does not exist on Windows. I find that rather strange and inconvenient.,,

Comment (EmilyBourne) [reply pyccel/codegen/utilities.py:65]: For pyccel these are specified as options in the help. For epyccel they are in the kwargs

**Pull request merged**